src/pages/PopupPage.js

```diff
@@ -104,21 +106,52 @@ export default function PopupPage({ opener }) {
     const request = requests[0];
     assert(
       request.method === 'signTransaction' ||
-      request.method === 'signAllTransactions',
+        request.method === 'signAllTransactions' ||
+        request.method === 'sign',
     );
 
-    let messages =
-      request.method === 'signTransaction'
-        ? [bs58.decode(request.params.message)]
-        : request.params.messages.map((m) => bs58.decode(m));
+    let messages, messageType;
+    switch (request.method) {
+      case 'signTransaction':
+        messages = [bs58.decode(request.params.message)];
+        messageType = 'tx';
+        break;
+      case 'signAllTransactions':
+        messages = request.params.messages.map((m) => ({
+          data: bs58.decode(m),
+          type: 'tx',
+        }));
+        messageType = 'tx';
+        break;
+      case 'sign':
+        if (typeof request.params.data === 'string') {
```

Comment:
The reason I detect the type here is to try to have a better UX when it's a text message, as I am then able in the popup to decide to display as human-readable text. If the input for this method is always  `UInt8Array` then I lose this ability as it's hard to tell if arbitrary data is some utf string or not.
But I think instead of relying directly on the type of `data`, we can have `data` always be a `UInt8Array` and have a second param like `encoding` that hint Sollet how it should be displayed to the user (e.g: `hex`, `utf8`...). I think that would be even better actually! I will start implementing it this way, but let me know if you think otherwise.

Reply:
Changed